frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/rbac.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  AccessReviewResourceAttributes,
+  K8sVerb,
+  SelfSubjectAccessReviewKind,
+} from '../../../extensions/console-types';
+import { ProjectModel, SelfSubjectAccessReviewModel } from '../../../models';
+import { k8sCreate } from '../../../utils/k8s/k8s-resource';
+import { getImpersonate } from '../../core/reducers/coreSelectors';
+import storeHandler from '../../storeHandler';
+import { useSafetyFirst } from '../safety-first';
+
+// Memoize the result so we only make the request once for each access review.
+// This does mean that the user will have to refresh the page to see updates.
+// Accept an `impersonateKey` parameter to include in the cache key even though
+// it's not used in the function body. (Impersonate headers are added
+// automatically by `k8sCreate`.) This function takes in the destructured
+// resource attributes so that the cache keys are stable. (`JSON.stringify` is
+// not guaranteed to give the same result for equivalent objects.)
+export const checkAccessInternal = _.memoize(
+  (
+    group: string,
+    resource: string,
+    subresource: string,
+    verb: K8sVerb,
+    name: string,
+    namespace: string,
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    impersonateKey: string,
+  ): Promise<SelfSubjectAccessReviewKind> => {
+    // Projects are a special case. `namespace` must be set to the project name
+    // even though it's a cluster-scoped resource.
+    const reviewNamespace =
+      group === ProjectModel.apiGroup && resource === ProjectModel.plural ? name : namespace;
+    const ssar: SelfSubjectAccessReviewKind = {
+      apiVersion: 'authorization.k8s.io/v1',
+      kind: 'SelfSubjectAccessReview',
+      spec: {
+        resourceAttributes: {
+          group,
+          resource,
+          subresource,
+          verb,
+          name,
+          namespace: reviewNamespace,
+        },
+      },
+    };
+    return k8sCreate(SelfSubjectAccessReviewModel, ssar);
+  },
+  (...args) => args.join('~'),
+);
+
+export const getImpersonateKey = (impersonate): string => {
+  const newImpersonate = impersonate || getImpersonate(storeHandler.getStore().getState());
+  return newImpersonate ? `${newImpersonate.kind}~{impersonate.user}` : '';
+};
+
+export const checkAccess = (
```

Comment:
would be nice to add jsdoc comments for the utils/hooks this file would expose from SDK